File: backend/app/application/dto/play_dto.py
from dataclasses import dataclass
from datetime import datetime
from uuid import UUID


@dataclass
class CreatePlayDTO:
    baby_id: UUID
    play_type: str
    started_at: datetime
    ended_at: datetime | None = None
    duration_minutes: int | None = None
    memo: str | None = None
    id: UUID | None = None  # 클라 생성 UUID(멱등 upsert)


@dataclass
class PlayResponseDTO:
    id: UUID
    baby_id: UUID
    play_type: str
    started_at: datetime
    ended_at: datetime | None
    duration_minutes: int | None
    memo: str | None
    created_at: datetime

File: zzippu_ios/zzippu/Data/Network/DTOs/PlayDTO.swift
// Data/Network/DTOs/PlayDTO.swift
// 서버 PlayResponse / PlayCreateRequest 대응

import Foundation

// MARK: - Response DTO

struct PlayResponseDTO: Decodable {
    let id: UUID
    let babyId: UUID
    let playType: String
    let startedAt: Date
    let endedAt: Date?
    let durationMinutes: Int?
    let memo: String?
    let createdAt: Date
}

// MARK: - Create Request DTO

struct PlayCreateRequestDTO: Encodable {
    let playType: String
    let startedAt: Date
    let endedAt: Date?
    let durationMinutes: Int?
    let memo: String?
}
